File: hlp_rams.py
def p(n):
	ans,q=0,2
	while n/q:
		ans+=n/q
		q=2*q
	return ans
t=input()
while t:
	t-=1
	n=input()
	o,e,pn=2,0,p(n)
	for k in range(1,n):
		if pn-p(k)-p(n-k):
			e+=1
		else:
			o+=1
	print '%d %d' %(e,o)

File: hlp_rams.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
ll pr(ll n)
{
	ll ans=1;
	while (n)
	{
		ans*=((n&1)+1);
		n>>=1;
	}
	return ans;
}
int main()
{
	int t;
	scanf("%d",&t);
	while(t--)
	{
		ll n,k;
		scanf("%lld",&n);
		k=pr(n);
		printf("%lld %lld\n", n+1-k,k);
	}
	return 0;
}